### PR TITLE
ci: change DATA_HOME to CACHE_HOME in basic-workflow-tests.sh

### PR DESCRIPTION
**Pull request by alimaredia**
Models cannot be served per changes from: https://github.com/instructlab/instructlab/pull/1762.

For example `ilab model serve` is looking for models in /tmp/tmp.XXXX/.local/share/instructlab/models/  but the model is being saved in /tmp/tmp.XXXX/.cache/instructlab/models/ via `ilab model download`.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: scripts/basic-workflow-tests.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ SCRIPTDIR=$(dirname "$0")
 export E2E_TEST_DIR
 # E2E_TEST_DIR="$(pwd)/__e2e_test"
 export CONFIG_HOME
-export DATA_HOME
+export CACHE_HOME
```
Comment (RobotSail) [inline scripts/basic-workflow-tests.sh:37]: Did you mean to remove `DATA_HOME`? It seems like you renamed all instances of `DATA_HOME` to now be `CACHE_HOME`.

File: scripts/basic-workflow-tests.sh [hunk 2]
```diff
@@ -227,12 +232,12 @@ test_train() {
     task Train the model
 
     if [ "$TRAIN_LIBRARY" -eq 1 ]; then
-        DATA=$(find "${DATA_HOME}"/instructlab/datasets -name 'messages_*' | head -n 1)
+        DATA=$(find "${CACHE_HOME}"/instructlab/datasets -name 'messages_*' | head -n 1)
```
Comment (RobotSail) [inline scripts/basic-workflow-tests.sh:235]: This should still be `DATA_HOME`

File: scripts/basic-workflow-tests.sh [hunk 3]
```diff
@@ -312,7 +317,7 @@ test_exec() {
     #   `ilab model convert` is only implemented for macOS with M-series chips for now
     #test_convert
 
-    test_serve "${DATA_HOME}/instructlab/models/ggml-model-f16.gguf"
+    test_serve "${CACHE_HOME}/instructlab/models/ggml-model-f16.gguf"
```
Comment (RobotSail) [inline scripts/basic-workflow-tests.sh:320]: This is the trained output if I'm not mistaken so I believe this should be under user data, but I could be wrong here. Maybe @mbestavros could clarify here.

**Review by RobotSail — COMMENTED** [review]
It seems like you renamed `DATA_HOME` -> `CACHE_HOME` which isn't entirely true. You'll need to have both `DATA_HOME` and `CACHE_HOME` defined here. 

I left comments indicating where we should still be using `DATA_HOME`.

Once that's addressed, we can merge this.

File: scripts/basic-workflow-tests.sh [hunk 4]
```diff
@@ -162,12 +162,17 @@ test_serve() {
         SERVE_ARGS+=("--model-path" "${model}")
     fi
     if [ "$BACKEND" = "vllm" ]; then
-        SERVE_ARGS+=("--model-path" "${DATA_HOME}/instructlab/models/instructlab/merlinite-7b-lab")
+        SERVE_ARGS+=("--model-path" "${CACHE_HOME}/instructlab/models/instructlab/merlinite-7b-lab")
     fi
 
     task Serve the model
+    # output serve log while ilab model serve is starting up
```
Comment (booxter) [inline scripts/basic-workflow-tests.sh:169]: probably doesn't belong here; I think you have the same change elsewhere?

File: scripts/basic-workflow-tests.sh [hunk 5]
```diff
@@ -275,12 +280,12 @@ test_exec() {
     test_serve
     PID=$!
 
-    test_chat
+    #test_chat
```
Comment (booxter) [inline scripts/basic-workflow-tests.sh:283]: obviously, these should be reverted back

File: scripts/basic-workflow-tests.sh (hunk 3 above)
Comment (alimaredia) [reply scripts/basic-workflow-tests.sh:317]: Looks like it should be $CACHE_HOME based on the change here: https://github.com/instructlab/instructlab/pull/1762/files#diff-ec289415187eaed5e0c70aac3bb11bc066ae949fa64b16a1dd786359954d06dfR123

Comment (mbestavros) [reply scripts/basic-workflow-tests.sh:317]: The general rule _for now_ is that any model we aren't downloading should be in the `checkpoints` directory, which is in `DATA_HOME`. (aka: `$DATA_HOME/checkpoints/model.gguf`)

File: scripts/basic-workflow-tests.sh [hunk 6]
```diff
@@ -47,7 +48,7 @@ export CONFIG_HOME
 # Arguments:
 #   None
 # Globals:
-#   DATA_HOME
```
Comment (booxter) [inline scripts/basic-workflow-tests.sh:50]: This should probably list both, not replace.

File: scripts/basic-workflow-tests.sh [hunk 7]
```diff
@@ -61,9 +62,10 @@ function init_e2e_tests() {
 
     CONFIG_HOME=$(python -c 'import platformdirs; print(platformdirs.user_config_dir())')
     DATA_HOME=$(python -c 'import platformdirs; print(platformdirs.user_data_dir())')
+    CACHE_HOME=$(python -c 'import platformdirs; print(platformdirs.user_cache_dir())')
     STATE_HOME=$(python -c 'import platformdirs; print(platformdirs.user_state_dir())')
     # ensure that our mock e2e dirs exist
-    for dir in "${CONFIG_HOME}" "${DATA_HOME}" "${STATE_HOME}"; do
+    for dir in "${CONFIG_HOME}" "${CACHE_HOME}" "${STATE_HOME}"; do
```
Comment (booxter) [inline scripts/basic-workflow-tests.sh:68]: should it create both?

Comment (RobotSail) [reply scripts/basic-workflow-tests.sh:68]: Yes

Comment (alimaredia) [reply scripts/basic-workflow-tests.sh]: Changed

**Review by booxter — APPROVED** [review]

**Review by RobotSail — APPROVED** [review]
LGTM

**Comment by alimaredia** [comment]
@booxter FYI the success on that last E2E run. It did not actually finish all the way through. I will kick off another one shortly.

**Comment by booxter** [comment]
@alimaredia [indeed](https://github.com/instructlab/instructlab/actions/runs/10083226932/job/27879432971). Why is the script not failing the run?..

**Comment by booxter** [comment]
This is completely untested, but I wonder if run_training error handling should be improved somehow. E.g. somewhere here: https://github.com/instructlab/training/pull/151

**Comment by alimaredia** [comment]
In the last run the training phase passed but since `$FULLTRAIN`  was is set to `0` the final ilab chat and serve weren't run (https://github.com/instructlab/instructlab/blob/main/scripts/basic-workflow-tests.sh#L304).

This does not test the very last code change in this PR. I'm going to kick off a A10x1 E2E job to see if that final path change can run.

**Pull request closed without merging**